### PR TITLE
[MO] - [system test] -> oauth scope

### PR DESCRIPTION
**Pull request by see-quick**
### Type of change

- Enhancement / new feature

### Description

This PR adding a test cases for verification of the OAuth scope parameter for clients (in this case the Kafka Connect) and also clientScope verification for KafkaClients. 

Maybe you ask yourself why there is no test case for the `KafkaBridge` or `KafkaMirrorMaker` component. The answer is simple. These components using the same mechanism so we would test the same paths in the operator code...that's why I decided to verify only KafkaConnect.
 
### Checklist

- [x] Write tests
- [x] Make sure all tests pass

### CODE REVIEW

File: systemtest/src/test/java/io/strimzi/systemtest/security/oauth/OauthScopeST.java [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+                .endSpec()
+            .build());
+
+        String kafkaConnectPodName = PodUtils.getPodsByPrefixInNameWithDynamicWait(namespaceName, clusterName).get(0).getMetadata().getName();
+        // kafka connect should fail because we do not specify audience but only scope
+        PodUtils.waitUntilPodIsInCrashLoopBackOff(namespaceName, kafkaConnectPodName);
+
+        // re-configuring Kafka listener to have client scope assigned to null
+        KafkaResource.replaceKafkaResourceInSpecificNamespace(clusterName, kafka -> {
+            List<GenericKafkaListener> audieListener = kafka.getSpec().getKafka().getListeners()
```
Comment (Frawless) [inline systemtest/src/test/java/io/strimzi/systemtest/security/oauth/OauthScopeST.java:112]: Maybe better name?

**Review by Frawless — APPROVED** [review]

File: systemtest/src/test/java/io/strimzi/systemtest/security/oauth/OauthScopeST.java [hunk 2]
```diff
@@ -0,0 +1,264 @@
...
+        final String producerName = OAUTH_PRODUCER_NAME + "-" + clusterName;
+        final String consumerName = OAUTH_CONSUMER_NAME + "-" + clusterName;
+        final String topicName = mapWithTestTopics.get(extensionContext.getDisplayName());
+
+        // SCOPE TESTING
+        resourceManager.createResource(extensionContext, KafkaClientsTemplates.kafkaClients(false, kafkaClientsName).build());
+        resourceManager.createResource(extensionContext, KafkaConnectTemplates.kafkaConnect(extensionContext, clusterName, clusterName, 1)
+            .withNewSpec()
+                .withReplicas(1)
+                .withBootstrapServers(KafkaResources.bootstrapServiceName(oauthClusterName) + ":" + audienceListenerPort)
```
Comment (mstruk) [inline systemtest/src/test/java/io/strimzi/systemtest/security/oauth/OauthScopeST.java:109]: `audienceListenerPort` -> `scopeListenerPort`

File: systemtest/src/test/java/io/strimzi/systemtest/security/oauth/OauthScopeST.java [hunk 3]
```diff
@@ -0,0 +1,264 @@
...
+        final String kafkaClientsName = mapWithKafkaClientNames.get(extensionContext.getDisplayName());
+        final String clusterName = mapWithClusterNames.get(extensionContext.getDisplayName());
+        final String producerName = OAUTH_PRODUCER_NAME + "-" + clusterName;
+        final String consumerName = OAUTH_CONSUMER_NAME + "-" + clusterName;
+        final String topicName = mapWithTestTopics.get(extensionContext.getDisplayName());
+
+        KafkaBasicExampleClients oauthInternalClientChecksJob = new KafkaBasicExampleClients.Builder()
+            .withProducerName(producerName)
+            .withConsumerName(consumerName)
+            .withBootstrapAddress(KafkaResources.bootstrapServiceName(oauthClusterName) + ":" + audienceListenerPort)
```
Comment (mstruk) [inline systemtest/src/test/java/io/strimzi/systemtest/security/oauth/OauthScopeST.java:151]: `audienceListenerPort` -> `scopeListenerPort`

File: systemtest/src/test/java/io/strimzi/systemtest/security/oauth/OauthScopeST.java [hunk 4]
```diff
@@ -0,0 +1,264 @@
...
```
Comment (mstruk) [inline systemtest/src/test/java/io/strimzi/systemtest/security/oauth/OauthScopeST.java:179]: audienceListenerPort -> scopeListenerPort

File: systemtest/src/test/java/io/strimzi/systemtest/security/oauth/OauthScopeST.java [hunk 5]
```diff
@@ -0,0 +1,264 @@
...
+
+        resourceManager.createResource(extensionContext, oauthInternalClientChecksJob.producerStrimzi().build());
+        // client should fail because we do not specify client-audience but only scope
+        ClientUtils.waitForClientTimeout(producerName, NAMESPACE, MESSAGE_COUNT);
+        JobUtils.deleteJobWithWait(NAMESPACE, producerName);
+
+        // rollback previous configuration
+        // re-configuring Kafka listener to have client scope assigned to 'test'
+        KafkaResource.replaceKafkaResourceInSpecificNamespace(oauthClusterName, kafka -> {
+            List<GenericKafkaListener> audienceListeners = kafka.getSpec().getKafka().getListeners()
```
Comment (mstruk) [inline systemtest/src/test/java/io/strimzi/systemtest/security/oauth/OauthScopeST.java:209]: `audience` -> `scope` in many places in this code block.

**Review by mstruk — APPROVED** [review]
Looks good, but I added some suggestions for improvements

**Review by scholzj — APPROVED** [review]

**Pull request merged**